### PR TITLE
Use a record-replay deterministic hash for Member types that contain ScriptWrappable objects

### PR DESCRIPTION
**Pull request by sgeisenh**
https://linear.app/replay/issue/RUN-1665/detect-and-automatically-make-all-instances-of-wtf-set-and-map

This change makes the hash functions that @klochek added the default for `Member`'s that contain subclasses of `ScriptWrappable`.

Once this and #545 are merged, I can go back through the containers where a record-replay deterministic hash was set manually and validate that the default hash should now have the behavior that we want.

### CODE REVIEW

**Review by kannanvijayan — APPROVED** [review]
LGTM

File: third_party/blink/renderer/platform/heap/member.h [hunk 1]
```diff
@@ -174,7 +174,8 @@ struct MemberHashRecordReplayId
             std::enable_if_t<WTF::IsAnyMemberType<Member>::value>* = nullptr>
   static unsigned GetHash(const Member& m) {
     if (recordreplay::IsRecordingOrReplaying()) {
-      int id = m.Get()->RecordReplayId();
+      T* ptr = m.Get();
+      int id = ptr->RecordReplayId();
```
Comment (bhackett1024) [inline third_party/blink/renderer/platform/heap/member.h:178]: Is this change necessary?  The earlier version is cleaner.

**Review by bhackett1024 — APPROVED** [review]
Nice!

**Comment by sgeisenh** [comment]
Not quite as clean as I thought, I ended up having to replace a couple of forward declarations with `#includes`. It doesn't seem like there are cyclic dependencies that were prevented by the forward declarations, so I'm not 100% sure why they were being used.
